### PR TITLE
Test case to check deployment of capp before transport starts

### PR DESCRIPTION
**Pull request by ThishaniLucas**
PR sent to ESB : https://github.com/wso2/product-esb/pull/654

### CODE REVIEW

File: integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/car/deployment/test/CARBON16113DeployArtifactsBeforeTransportStarsTestCase.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    @Test(groups = { "wso2.esb"}, description = "Testing whether CApp is deployed before transport starts")
+    public void carReDeploymentTest() throws Exception {
+        logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
+        logViewerClient.clearLogs();
+        uploadCapp("car-deployment-before-tranaport-start-test_1.0.0.car"
+                , new DataHandler(new URL("file:" + File.separator + File.separator + getESBResourceLocation()
+                        + File.separator + "car" + File.separator +
+                        "car-deployment-before-tranaport-start-test_1.0.0.car")));
+        //thread sleep to give some time for the capp to get deployed
+        Thread.sleep(5000);
```
Comment (sdkottegoda) [inline integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/car/deployment/test/CARBON16113DeployArtifactsBeforeTransportStarsTestCase.java]: Doesn't the method, uploadApp return after the verification of the deployment of the Capp? If so, I think this should be handled inside the uploadCapp itself, or else we'll have different thread sleeps in different test cases resulting in inconsistencies

File: integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/car/deployment/test/CARBON16113DeployArtifactsBeforeTransportStarsTestCase.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    @Test(groups = { "wso2.esb"}, description = "Testing whether CApp is deployed before transport starts")
+    public void carReDeploymentTest() throws Exception {
+        logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
+        logViewerClient.clearLogs();
+        uploadCapp("car-deployment-before-tranaport-start-test_1.0.0.car"
+                , new DataHandler(new URL("file:" + File.separator + File.separator + getESBResourceLocation()
+                        + File.separator + "car" + File.separator +
+                        "car-deployment-before-tranaport-start-test_1.0.0.car")));
```
Comment (sdkottegoda) [inline integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/car/deployment/test/CARBON16113DeployArtifactsBeforeTransportStarsTestCase.java]: Line wrapping should happen before an operator

File: integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/car/deployment/test/CARBON16113DeployArtifactsBeforeTransportStarsTestCase.java (hunk 1 above)
Comment (ThishaniLucas) [reply integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/car/deployment/test/CARBON16113DeployArtifactsBeforeTransportStarsTestCase.java]: Yes. uploadApp is a blocking method. Removed the thread sleep.

File: integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/car/deployment/test/CARBON16113DeployArtifactsBeforeTransportStarsTestCase.java (hunk 2 above)
Comment (ThishaniLucas) [reply integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/car/deployment/test/CARBON16113DeployArtifactsBeforeTransportStarsTestCase.java]: fixed

**Pull request merged**